### PR TITLE
Rename some sizevars methods for clarity

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #105714
* __->__ #105585

The guard functions require you to ALREADY KNOW that a particular
condition holds.  If you don't know (you want to guard on an expression
being a particular value, and then get access to that value), use
the evaluate functions.

I renamed the functions that don't abide by this:

```
guard_min -> evaluate_min
guard_max (deleted, no uses)
guard_static_shape -> evaluate_static_shape
guard_static_shapes -> evaluate_static_shapes
```

Some added comments.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

**Review by voznesenskym — APPROVED** [review]

File: torch/_inductor/sizevars.py [hunk 1]
```diff
@@ -317,7 +334,11 @@ def guard_lt(self, left: Expr, right: Expr) -> None:
             return
         assert self.shape_env.evaluate_expr(sympy.Lt(left, right))
 
-    def guard_min(self, left: Expr, right: Expr) -> Expr:
+    # The evaluate functions evaluate some symbolic sympy expression
+    # (NB: not necessarily an Expr) and return what the concrete result
+    # is, guarding on the expression being that result
+
```
Comment (voznesenskym) [inline torch/_inductor/sizevars.py:340]: nit newline

File: torch/_inductor/sizevars.py [hunk 2]
```diff
@@ -307,6 +311,19 @@ def statically_known_multiple_of(self, numerator: Expr, denominator: Expr) -> bo
         expr = sympy.Eq(numerator % denominator, 0)
         return self.is_expr_static_and_true(expr)
 
+    # The guard functions require you to ALREADY KNOW that a particular
+    # condition holds.  If you don't know (you want to guard on an expression
+    # being a particular value, and then get access to that value), use
+    # the evaluate functions.
+
```
Comment (voznesenskym) [inline torch/_inductor/sizevars.py:318]: nit newline? also maybe we should just always go through speculation? requiring that they already know can be an assert at the callsite...?

File: torch/_inductor/sizevars.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/sizevars.py:335]: Intentional, there is more than one evaluate function, deleting the newline associates it with the wrong function

**Pull request closed without merging**